src/storage/src/sink/kafka.rs

```diff
@@ -1506,3 +1439,17 @@ where
 
     output_stream
 }
+
+#[derive(Serialize, Deserialize)]
+/// This struct is emitted as part of a transactional produce, and captures the information we
+/// need to resume the Kafka sink at the correct place in the sunk collection. (Currently, all
+/// we need is the timestamp... this is a record to make it easier to add more metadata in the
+/// future if needed.) It's encoded as JSON to make it easier to introspect while debugging, and
+/// because we expect it to remain small.
+///
+/// Unlike the old consistency topic, this is not intended to be a user-facing feature; it's there
+/// purely so the sink can maintain its transactional guarantees. Any future user-facing consistency
+/// information should be added elsewhere instead of overloading this record.
+struct ProgressRecord {
+    timestamp: Timestamp,
```

Comment:
Bikeshed opportunity: other places in the code call this a `transaction` or `transaction_id`. I have a small preference for `timestamp`, but only a small one if others feel strongly.

Reply:
it's only in the (kafka) sink code that we call it transaction, right? I would also lean towards renaming all those to timestamp.

Reply:
I'll follow up with the code-level renames. @aljoscha also pointed out that the messages in the output topic refer to this as the "transaction" as well, which we should consider changing as a followup.